File: 101-symmetric-tree/101-symmetric-tree.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def isSymmetric(self, root: Optional[TreeNode]) -> bool:
        def p(node1, node2):
            if not node1 and not node2:
                return True
            elif not node1 or not node2:
                return False
            elif node1.val != node2.val:
                return False

            return p(node1.left, node2.right) and p(node1.right, node2.left)

        if not root:
            return True

        return p(root.left, root.right)

                
                
                
                 
            
        

File: 101-symmetric-tree/101-symmetric-tree.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val, left, right) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.left = (left===undefined ? null : left)
 *     this.right = (right===undefined ? null : right)
 * }
 */
/**
 * @param {TreeNode} root
 * @return {boolean}
 */
var isSymmetric = function(root) {
    
    const p = function(node1,node2){
        
        if(node1 == null & node2 == null){
            return true
        }
        else if(node1 == null | node2 == null){
            return false
        }
        else if(node1.val != node2.val){
            return false
        };
        return p(node1.left,node2.right) & p(node1.right,node2.left)
    };
    return p(root.left,root.right)
    
};